### PR TITLE
Feature 49 sale product return

### PR DESCRIPTION
**Pull request by dimitrov93**

### CODE REVIEW

File: src/views/Products/ProductsDetailsById.vue [hunk 1]
```diff
@@ -158,4 +166,24 @@ const postPicture = async (id, image) => {
     snackbarProvider.showErrorSnackbar(error?.response?.data?.error);
   }
 };
+
+const submitReturn = () => {
+  const confirm = window.confirm(
+    "Are you sure that you would like to return this sale?"
```
Comment (VladoKat) [inline src/views/Products/ProductsDetailsById.vue:172]:
```suggestion
    "Are you sure that you would like to return this product?"
```

File: src/services/HttpClientService.js [hunk 2]
```diff
@@ -142,3 +142,7 @@ export async function updateUser(id, userData) {
 export async function transferProduct(productId, recipientId) {
   return await updateData(`/products/${productId}/transfer/${recipientId}`);
 }
+
+export async function saleReturn(productId) {
```
Comment (VladoKat) [inline src/services/HttpClientService.js:146]: Let's be a bit clearer and rename this to `productReturn`

File: src/store/modules/sales.js [hunk 3]
```diff
@@ -33,6 +34,9 @@ export default {
     async postSale({ commit }, data) {
       await postSale(data);
     },
+    async returnSale({ commit }, productId) {
```
Comment (VladoKat) [inline src/store/modules/sales.js:37]: same with this. We are returning a product

File: src/views/Sales/SaleDetails.vue [hunk 4]
```diff
@@ -24,6 +24,12 @@
         <template v-slot:item.owner="{ item }">
           <user-tool-tip :user="item.owner" @click.stop />
         </template>
+
+        <template v-slot:item.return="{ item }">
+          <v-btn variant="plain" @click.stop="submitReturn(item)">
+            <v-icon size="25">mdi-backup-restore</v-icon>
```
Comment (VladoKat) [inline src/views/Sales/SaleDetails.vue:30]: I don't like the icon so I went ahead and looked at some. Should we go with cart-remove, cart-arrow-up or... cash-remove, or cash-refund. Take a look. I couldn't decide among these. Take your pick

**Review by VladoKat — CHANGES_REQUESTED** [review]
Changes look good overall. I was thinking whether we can have the return-product button extracted in a component? I see we are repeating the logic in the 2 views currently. I realize the button looks different in the 2 views but maybe there is something we can do about it?

File: src/views/Products/ProductsDetailsById.vue (hunk 1 above)
Comment (dimitrov93) [reply src/views/Products/ProductsDetailsById.vue:172]: fixed

File: src/services/HttpClientService.js (hunk 2 above)
Comment (dimitrov93) [reply src/services/HttpClientService.js:146]: fixed

File: src/store/modules/sales.js (hunk 3 above)
Comment (dimitrov93) [reply src/store/modules/sales.js:37]: fixed

File: src/views/Sales/SaleDetails.vue (hunk 4 above)
Comment (dimitrov93) [reply src/views/Sales/SaleDetails.vue:30]: cart-remove looks the most suitable for me

File: src/components/Button/ReturnProductButton.vue [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<template>
+    <v-btn
+      v-if="!isSalesPage && isPartOfProduct"
+      color="#673AB7"
+      @click.stop="submitReturn"
+      :size="isMediumScreen() ? 'x-small' : 'default'"
+      >Sale Return</v-btn
```
Comment (VladoKat) [inline src/components/Button/ReturnProductButton.vue:7]:
```suggestion
      >Product Return</v-btn
```

File: src/views/Sales/SaleDetails.vue [hunk 6]
```diff
@@ -45,7 +51,31 @@ const currentSale = computed(() => store.getters["sales/getSaleById"](saleId));
...
+
+const submitSaleReturn = (productId) => {
+  try {
+    store.dispatch("sales/returnProduct", productId);
+    snackbarProvider.showSuccessSnackbar("Product has been returned from sale");
+    router.go(-1);
+  } catch (error) {
+    snackbarProvider.showErrorSnackbar(error?.response?.data?.error);
+  }
+};
```
Comment (VladoKat) [inline src/views/Sales/SaleDetails.vue:78]: Do we still need this? Isn't it exported to the button?

**Review by VladoKat — CHANGES_REQUESTED** [review]

File: src/views/Sales/SaleDetails.vue [hunk 7]
```diff
@@ -45,7 +49,11 @@ const currentSale = computed(() => store.getters["sales/getSaleById"](saleId));
 const saleProducts = currentSale.value.products.map((product) =>
   formatProducts(product)
 );
-const ownerColumn = computed(() => [store.state.products.tableColumnOwner]);
+const ownerColumn = computed(() => [
```
Comment (VladoKat) [inline src/views/Sales/SaleDetails.vue:52]: Last thing: this is now `additionalColumns` (it's not jus the owner column)

**Review by VladoKat — CHANGES_REQUESTED** [review]

File: src/views/Sales/SaleDetails.vue (hunk 7 above)
Comment (dimitrov93) [reply src/views/Sales/SaleDetails.vue]: done

**Review by VladoKat — APPROVED** [review]

**Pull request merged**